Doc/library/io.rst

```diff
@@ -826,8 +823,8 @@ than raw I/O does.
 
       Write the :term:`bytes-like object`, *b*, and return the
       number of bytes written.  When in non-blocking mode, a
-      :exc:`BlockingIOError` is raised if the buffer needs to be written out but
-      the raw stream blocks.
+      :exc:`BlockingIOError` with :attr:`BlockingIOError.characters_written` set
```

Comment:
characters_written is fairly well tested in this case (https://github.com/python/cpython/blob/a7f317d7300235d9efff5a0350b1bae14720d42f/Lib/test/test_io.py#L1918-L1943)